everything-presence-mmwave-configurator/www/script.js

```diff
@@ -72,7 +72,36 @@ document.addEventListener("DOMContentLoaded", () => {
   function calculateOffsetY() {
     let absAngle = Math.abs(installationAngle);
     if (absAngle <= 30) offsetY = 0;
-    else offsetY = detectionRange * Math.sin((absAngle - 30) * Math.PI / 180);
+    else offsetY = detectionRange * Math.sin(((absAngle - 30) * Math.PI) / 180);
+  }
+
+  /// Returns the entity's state converted from whatever unit is configured in the UI converted to millimeters
+  function getEntityStateMM(entity) {
```

Comment:
We have to round the values that this function returns. I've adapted the function below:

```JS
/// Returns the entity's state converted from whatever unit is configured in the UI converted to millimeters
function getEntityStateMM(entity) {
  const state = entity ? parseFloat(entity.state) || 0 : 0;
  let result = state;
  // cm, in, ft, km, m, mi, nmi, yd and mm are supported in home assistant.
  switch (entity.attributes.unit_of_measurement) {
    case 'mm':
      break; // Avoid checking every unit for the most common case
    case "in":
      result = state * 25.4; // Convert inches to millimeters
      break;
    case "ft":
      result = state * 304.8; // Convert feet to millimeters
      break;
    case "km":
      result = state * 1000000; // Convert kilometers to millimeters
      break;
    case "in":
      result = state * 1000; // Convert meters to millimeters
      break;
    case "in":
      result = state * 1.609e6; // Convert miles to millimeters
      break;
    case "in":
      result = state * 1.852e6; // Convert nautical miles to millimeters
      break;
    case "in":
      result = state * 914.4; // Convert yards to millimeters
      break;
  }
  return Math.round(result);
}
```

Reply:
Thanks for testing!
I Integrated your suggestion- I did fixup the last couple cases for miles/nautical miles/yards that were still case 'in' . This looks like it's working well on my end though.

Reply:
Ups 😅 thanks!